pkgs/by-name/ro/rovium/package.nix

```
@@ -0,0 +1,141 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  autoPatchelfHook,
+  makeWrapper,
+  dpkg,
+  glib,
+  gtk3,
+  nspr,
+  nss,
+  cups,
+  dbus,
+  atk,
+  at-spi2-atk,
+  at-spi2-core,
+  cairo,
+  pango,
+  libX11,
+  libXcomposite,
+  libXdamage,
+  libXext,
+  libXfixes,
+  libXrandr,
+  libxcb,
+  libxkbcommon,
+  libdrm,
+  mesa,
+  libGL,
+  libxkbfile,
+  libsecret,
+  expat,
+  alsa-lib,
+  systemd,
+  fontconfig,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "rovium";
+  version = "0.4.0";
+
+  src = fetchurl {
+    url = "https://github.com/rovium/rovium-beta/releases/download/v${finalAttrs.version}/rovium-${finalAttrs.version}-amd64.deb";
+    hash = "sha256-JHFD82MSyvQMYRzzLBFdO8G1RHUKU5sHk5qgYyMqWVE=";
+  };
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    makeWrapper
+    dpkg
+  ];
+
+  buildInputs = [
+    stdenv.cc.cc.lib
+    glib
+    gtk3
+    nspr
+    nss
+    cups
+    dbus
+    atk
+    at-spi2-atk
+    at-spi2-core
+    cairo
+    pango
+    libX11
+    libXcomposite
+    libXdamage
+    libXext
+    libXfixes
+    libXrandr
+    libxcb
+    libxkbcommon
+    libdrm
+    mesa
+    libGL
+    libxkbfile
+    libsecret
+    expat
+    alsa-lib
+    systemd
+    fontconfig
+  ];
+
+  autoPatchelfIgnoreMissingDeps = [ "libc.musl-x86_64.so.1" ];
+
+  unpackPhase = ''
+    runHook preUnpack
+    dpkg-deb -x $src .
+    runHook postUnpack
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/opt
+
+    cp -r opt/Rovium $out/opt/
+
+    makeWrapper $out/opt/Rovium/rovium $out/bin/rovium \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath finalAttrs.buildInputs}" \
+      --set ELECTRON_ENABLE_LOGGING 0 \
+      --set ELECTRON_NO_UPDATER 1 \
+      --set DBUS_SESSION_BUS_ADDRESS "unix:path=/run/user/\$(id -u)/bus" \
+      --add-flags "--no-sandbox" \
+      --add-flags "--ozone-platform=x11" \
+      --add-flags "--disable-update" \
+      --add-flags "--disable-component-update" \
+      --add-flags "--disable-breakpad" \
+      --add-flags "--disable-background-networking" \
+      --add-flags "--enable-features=UseOzonePlatform"
+
+    if [ -f usr/share/applications/rovium.desktop ]; then
+      install -D --mode=644 usr/share/applications/rovium.desktop \
+        --target-directory=$out/share/applications
+      substituteInPlace $out/share/applications/rovium.desktop \
+        --replace-fail "/opt/Rovium/rovium" "$out/bin/rovium"
+    fi
+
+    if [ -d usr/share/icons/hicolor/512x512/apps ]; then
+      cp usr/share/icons/hicolor/512x512/apps/* $out/share/icons/hicolor/512x512/apps/ || true
+    fi
+
```

```suggestion

    substituteInPlace usr/share/applications/rovium.desktop \
      --replace-fail "/opt/Rovium/rovium" "$out/bin/rovium"

    install -D --mode=644 usr/share/applications/rovium.desktop \
      --target-directory=$out/share/applications

    install -D --mode=644 usr/share/icons/hicolor/512x512/apps/rovium.png \
      --target-directory $out/share/icons/hicolor/512x512/apps

    cp --recursive opt $out

    makeWrapper $out/opt/Rovium/rovium $out/bin/rovium \
      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath finalAttrs.buildInputs}" \
      --set ELECTRON_ENABLE_LOGGING 0 \
      --set ELECTRON_NO_UPDATER 1 \
      --set DBUS_SESSION_BUS_ADDRESS "unix:path=/run/user/\$(id -u)/bus" \
      --add-flags "--no-sandbox" \
      --add-flags "--ozone-platform=x11" \
      --add-flags "--disable-update" \
      --add-flags "--disable-component-update" \
      --add-flags "--disable-breakpad" \
      --add-flags "--disable-background-networking" \
      --add-flags "--enable-features=UseOzonePlatform"

```

Cleanup (opinionated, non-blocking)